packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/index.tsx

```diff
@@ -0,0 +1,249 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import clsx from 'clsx';
+import { useTranslation } from 'react-i18next';
+import { Collapse } from '@mui/material';
+import styles from './styles.module.scss';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowUp';
+import { ReactComponent as SwapIcon } from '../../../../assets/images/swap.svg';
+import TextButton from '../../../Form/Button/TextButton';
+import Switch from '../../../Form/Switch';
+import Unit from '../../../Form/Unit';
+import NumberInput from '../../../Form/NumberInput';
+import {
+  location_area,
+  soilAmounts,
+  soilAmountsVolume,
+  applicationRateWeight,
+  applicationRateVolume,
+} from '../../../../util/convert-units/unit';
+import { TASK_PRODUCT_FIELD_NAMES, type TaskProductFormFields } from '../types';
+import { AreaApplicationSummary } from './AreaApplicationSummary';
+import { useQuantityApplicationRate } from './useQuantityApplicationRate';
+
+const SOIL_AMENDMENT_UNITS = {
+  WEIGHT: {
+    units: soilAmounts, // original
+  },
+  VOLUME: {
+    units: soilAmountsVolume,
+  },
+  APPLICATION_RATE_WEIGHT: {
+    units: applicationRateWeight,
+  },
+  APPLICATION_RATE_VOLUME: {
+    units: applicationRateVolume,
+  },
+};
+
+export interface Location {
+  type: string;
+  total_area: number;
+  total_area_unit: 'm2' | 'ha' | 'ft2' | 'ac';
+}
+
+export type QuantityApplicationRateProps = {
+  productId: number | string;
+  isReadOnly: boolean;
+  system: 'metric' | 'imperial';
+  location: Location;
+  defaultValues?: TaskProductFormFields;
+};
+
+const QuantityApplicationRate = ({
+  productId,
+  isReadOnly,
+  system, // measurementSelector
+  location,
+  defaultValues,
+}: QuantityApplicationRateProps) => {
+  const { t } = useTranslation();
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const toggleExpanded = () => setIsExpanded((prev) => !prev);
+
+  const { total_area, total_area_unit, type } = location;
+
+  const { control, getValues, setValue, register, watch } = useForm<TaskProductFormFields>({
+    defaultValues: {
+      [TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION]: 100,
+      ...defaultValues,
+    },
+  });
+
+  /* Control of relationship between quantity, area, and application rate */
+  const {
+    onQuantityChange,
+    onApplicationRateChange,
+    onPercentLocationChange,
+    isWeight,
+    toggleMeasure,
+    previewStringValue,
+    previewStringUnit,
+  } = useQuantityApplicationRate({
+    total_area,
+    total_area_unit,
+    system,
+    watch,
+    setValue,
+    getValues,
+  });
+
+  const percent_of_location = watch(TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION);
+
+  /* Label for weight/volume switch */
+  const formatLabel = (labelKey: string, shouldBeBold: boolean) => {
```

Comment:
I think we can assume Loic makes things that we want to be consistent in the future. And if he changes them we can then ask about the change. But yah only if you want to ! It would clean up this file a bit so it seemed nice to move out.